src/mainwindow.cpp

```diff
@@ -705,10 +697,16 @@ void MainWindow::compileAndRun()
     compile();
 }
 
-void MainWindow::formatSource()
+void MainWindow::formatSource(bool selectionOnly, bool logOnNoChange)
 {
     LOG_INFO("Requested code format");
-    formatter->format(editor, filePath, language, true, true);
+    Extensions::CodeFormatter *formatter;
```

Comment:
Any special reasons for heap allocation? It gets deleted as function ends. So why not make it local variable instead?

Reply:
How do you make it a local variable? `CodeFormatter` is an abstract class.

Reply:
Actually I can use two `formatter->format();`s. Do you think that's better?

Reply:
```cpp

    if (language == "Python")
        Extensions::YAPFormatter(editor, language, selectionOnly, logOnNoChange, log, this).format();
    else
        Extensions::ClangFormatter(editor, language, selectionOnly, logOnNoChange, log, this).format();
```

It seems good :thinking:

Reply:
> It seems good 

Sure it does and it's performance improvement too. Creating objects on heap and destroying them frequently is expensive. `CodeFormatter` just serves as base class for other formatter.